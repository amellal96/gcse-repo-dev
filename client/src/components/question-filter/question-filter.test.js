import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import { mount, shallow } from 'enzyme';

import QuestionFilter from './question-filter.component';

jest.mock('../../actions/questions');

const mockStore = configureStore([thunk]);

describe('QuestionFilter component', () => {
    let store;
    let wrapper;
    let mockChangeExamBoards;
    let mockRefreshPage;
    let mockChangeDifficulty;

    beforeEach(() => {
        store = mockStore({
            question: []
        });

        mockChangeExamBoards = jest.fn();
        mockRefreshPage = jest.fn();
        store.dispatch = jest.fn();
        mockChangeDifficulty = jest.fn();

        const mockProps = {
            store,
            refreshPage: mockRefreshPage,
            changeExamBoards: mockChangeExamBoards,
            changeDifficulty: mockChangeDifficulty
        }
        
        wrapper = mount(<Provider store={store}>
            <QuestionFilter {...mockProps} />
        </Provider>)
    });

    // it('expect changeExamBoards to be called', () => {
    //     const eMock = { preventDefault: jest.fn(), target: 'fake target', checked: true };
        
    //     expect(mockChangeExamBoards).not.toHaveBeenCalled();

    //     wrapper.instance().changeExamBoards = jest.fn();

    //     wrapper.find('.option-container')
    //         .at(0)
    //         .simulate('change', eMock);

    //     wrapper.find('.question-option-container')
    //         .at(0)
    //         .simulate('change', eMock);

    //     wrapper.find('.question-option')
    //         .at(0)
    //         .simulate('change', eMock);

    //     expect(mockChangeExamBoards).toHaveBeenCalled();
    // });

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot(); 
    });

    it('should render option fields', () => {
        expect(wrapper.find('.option-container')).toHaveLength(4);
    });

    // it('should check that checkbox is checked', () => {
    //     const eMock = { preventDefault: jest.fn(), target: 'fake target', checked: true };

    //     wrapper.find('.question-option')
    //         .at(2)
    //         .simulate('change', eMock);

    //     const checkbox = wrapper.find('.question-option').at(2);

    //     checkbox.simulate('change', eMock);

    //     expect(checkbox.checked).toEqual(true);
    // });

    it('check reload page occurs', () => {
        const mockEvent = { preventDefault: jest.fn() };

        window.location.reload = jest.fn();

        wrapper.find('.btn.btn-primary')
            .simulate('click', mockEvent);

        expect(window.location.reload).toHaveBeenCalled();
    });
});