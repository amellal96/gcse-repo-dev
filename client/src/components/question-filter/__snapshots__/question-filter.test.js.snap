// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionFilter component should match the snapshot 1`] = `"<div class=\\"container\\"><h3>Filter</h3><form class=\\"option\\"><h5>Exam Boards</h5><div class=\\"option-container\\"><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"AQA\\" value=\\"AQA\\"> AQA</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Edexcel\\" value=\\"Edexcel\\"> Edexcel</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"OCR\\" value=\\"OCR\\"> OCR</div><div class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"WJEC\\" value=\\"WJEC\\"> WJEC</div></div><h5>Topics</h5><div class=\\"option-container\\"><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Numbers\\" value=\\"Number\\"> Number </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Fractions\\" value=\\"Algebra\\"> Algebra </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Ratio, Proportion and Rates of change\\" value=\\"Ratio, Proportion and Rates of change\\"> Ratio, Proportion and Rates of change </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Geometry and Measures\\" value=\\"Geometry and Measures\\"> Geometry and Measures </div><div class=\\"question-option-container\\"> <input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Algebra\\" value=\\"Statistics &amp; Probability\\"> Statistics &amp; Probability </div></div><h5>Difficulty</h5><div class=\\"option-container\\"><label class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">From</label><input name=\\"min\\" class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"><label name=\\"max\\" class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">To</label><input class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"></div><h5>Marks</h5><div class=\\"option-container\\"><label class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">From</label><input name=\\"min\\" class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"><label name=\\"max\\" class=\\"col-form-label col-form-label-sm\\" for=\\"inputSmall\\">To</label><input class=\\"form-control form-control-sm filter\\" type=\\"text\\" id=\\"inputSmall\\"></div><button type=\\"button\\" class=\\"btn btn-primary\\">Clear</button></form></div>"`;
